### ISSUE TITLE
Posterize && GUI?

### ISSUE THREAD

**Issue by zvezdochiot** [plzombie/depress#10]
Hi @plzombie .

:question: The posterization algorithm will be effective only if you set the color set (palette) manually in the GUI. Algorithms for automatically compiling a palette on a small number of colors are practically useless. Any plans to embed such a tool in GUI?

**Comment by plzombie** [comment]
Hi @zvezdochiot . Posterization for small number of colors is quite ok for screenshots. I don't see a problem here.
![изображение](https://user-images.githubusercontent.com/1690651/234463128-d1a85dff-24fe-4bf4-80be-050778f8b493.png)
About gui. I already started project depressed (which stands for depress editor). I will finish it someday, somehow

**Comment by zvezdochiot** [comment]
Hi @plzombie .

Приличия соблюдены, теперь можно и по-нашенски.

О чём  я, собственно, речь веду. А, собственно, вот о чём. При всех моих манипуляциях со сканами, только отдельный класс материалов удавалось автоматически постеризовать (noteshrink, собственно). Но в большинстве случаев - это набор манипуляций в GIMP (различные фильтры, пороги и постеризации с злоупотреблением выделения), в том числе "Замена цвета", либо создание файла палитры в том же самом GIMP и применение его к изображению через NetPbm. Короче, без ручного составления палитры результат нулевой или близкий к нему. Такие вот дела.

У тебя есть "GUI-механизм" выбора цветов, которые надо применить к изображению?

**Comment by plzombie** [comment]
Нет. Я вообще думал libimagequant прикрутить, которая именно универсальная. Но она под gnu gpl. В любом случае, если ты сейчас подсунешь в прогу изображение уже с требуемой палитрой, она оставит её как есть (в режиме quant)

**Comment by zvezdochiot** [comment]
@plzombie say:
>libimagequant прикрутить, которая именно универсальная. 

Ничего в ней особо универсального. Просто немного проработаней друх (известно откуда руки растут). Но на малом количестве цветов не особо отличается от других, меньше 8-ми цветов запрашивать с неё не рекомендую.

 @plzombie say:
>изображение уже с требуемой палитрой 

А откуда ж оно вдруг возьмётся то? Обычно наблюдается наличие отсутствия.

**Comment by zvezdochiot** [comment]
Hi @plzombie . 

Ежели ещё есть желание похимичить с малоцветными масками (Fg), выложил слегка подрихтованный [pnmtodjvurle](https://github.com/ImageProcessing-ElectronicPublications/pnmtodjvurle): учитывая траблы с кодировками/перекодировками и режимами консоли винды, опционально разрешил ввод/вывод не через `stdin`/`stdout`, а через файлы:
```shell
pnmtodjvurle foreground.pnm foreground.rle
```
Единственное, забыл добавить проверку дескрипторов после этого выбора. Позже добавлю.

PS: В 1.1 всё пофиксил.

**Comment by zvezdochiot** [comment]
Hi @plzombie .

Ежели нужен k-means на Qt, то пожайлуста: [STEX: 0.2023.08.22: kmeans](https://github.com/ImageProcessing-ElectronicPublications/scantailor-experimental/releases/tag/0.2023.08.22). (разделение на BG и FG с помощью BW-маски).

**Comment by zvezdochiot** [comment]
Hi @plzombie .

[Здесь (Ru.Board)](http://forum.ru-board.com/topic.cgi?forum=5&topic=51044&start=220#11) гуишку для DjVuLibre просят. Depressed - это случаем не "оно"? Только в релизах её не видно.

**Comment by plzombie** [comment]
@zvezdochiot depressed - это по сути расширенный gui для depress. С возможностью ставить индивидуальные настройки для каждой страницы и с сохранением проекта в xml. Там нельзя, как они хотят, открыть существующий файл и поменять ориентацию у страницы. Ну и потом, некоторых фич вроде добавления нескольких файлов за раз или прогресс бара там нет пока.

**Comment by plzombie** [comment]
Поэтому я и не выкладываю бинарники

**Comment by plzombie** [comment]
Как-то так сейчас выглядит
![изображение](https://github.com/plzombie/depress/assets/1690651/13c40db9-a696-4ff0-ba85-90db532422e3)

**Comment by zvezdochiot** [comment]
Hi @plzombie .

Да мне то что? Мне без надобности. Просто не знал, что ты уже пользователей "прощупал" и решил, что не стоит. Ну так тому и быть, благодарности то всё-равно нет и быть не может. Так что твоё решение обосновано.
